### DISCUSSION TITLE
Migrating legacy Code to either InstalledCode or PortableCode

### DISCUSSION THREAD

**Discussion by ahkole** [aiidateam/aiida-core#5861] [Q&A]
Version 2.1.0 split the functionality of codes into two new classes `InstalledCode` and `PortableCode`. In the release notes (https://aiida.readthedocs.io/projects/aiida-core/en/stable/reference/_changelog.html#improved-interface-for-creating-codes) it is mentioned that existing codes should be automatically migrated to one of these new classes. However, after upgrading my installation from 2.0.4 to the newest version (2.2.1) according to the steps in the documentation (https://aiida.readthedocs.io/projects/aiida-core/en/stable/howto/installation.html#updating-your-installation) all my existing codes are now of type `aiida.orm.nodes.data.code.legacy.Code` and not one of the new classes. Do I have to perform some additional steps to migrate my existing codes? Or did something go wrong in my upgrade process?

**Comment by unkcpz** [comment]
Hi @ahkole, apologize for late reply. The `aiida.orm.nodes.data.code.legacy.Code` is for backward compatibility purposes, so everything is fine in your setup.
You can leave it as it is, using the old code node should be fine. You can also setup a new code with `InstalledCode`.
